### PR TITLE
Implemented dublicated properties substitution

### LINKED ISSUE #17: Properties are not expanded when mentioned multiple times in value
In original version of plugin there is a bug: https://github.com/mojohaus/properties-maven-plugin/issues/37
We have `PropertyResolver` class that is responsible to perform the properties lookup. This ticket showed that the design idea of storing to-be expanded part of value of another property in `unversolved` value is wrong. We need to refactor this class to make a recursive searches for property value

### PR DESCRIPTION
**Pull request by mipo256**
Closes #17

### CODE REVIEW

File: src/main/java/io/polivakha/mojo/properties/ReadPropertiesMojo.java [hunk 1]
```diff
@@ -236,7 +236,9 @@ private void resolveProperties() throws MojoExecutionException, MojoFailureExcep
         Properties projectProperties = project.getProperties();
 
         for (Object key : projectProperties.keySet()) {
-            projectProperties.setProperty( (String) key, getPropertyValue( (String) key, projectProperties, environment ) );
+            if (key.equals("first.property")) {
```
Comment (serezakorotaev) [inline src/main/java/io/polivakha/mojo/properties/ReadPropertiesMojo.java]: what is that?))

**Review by temalead — APPROVED** [review]

File: src/main/java/io/polivakha/mojo/properties/PropertyResolver.java [hunk 2]
```diff
@@ -33,37 +39,44 @@
...
-     * @param properties project properties
+     * @param mavenProjectProperties project properties
      * @param environment environment variables
-     * @return resolved property value
+     * @return resolved property value, or property placeholder, if it was not resolved
      * @throws IllegalArgumentException when properties are circularly defined
      */
-    public String getPropertyValue(String key, Properties properties, Properties environment) {
-        String valueFromProperties = properties.getProperty(key);
+    public String getPropertyValue(String key, Properties mavenProjectProperties, Properties environment, CircularDefinitionPreventer circularDefinitionPreventer) {
```
Comment (serezakorotaev) [inline src/main/java/io/polivakha/mojo/properties/PropertyResolver.java]: Maybe private?)

Comment (serezakorotaev) [reply src/main/java/io/polivakha/mojo/properties/PropertyResolver.java]: ok

**Review by serezakorotaev — APPROVED** [review]
Good job!

**Pull request merged**